runelite-client/src/main/java/net/runelite/client/plugins/inventorygrid/InventoryGridOverlay.java

```diff
@@ -69,48 +90,171 @@ private InventoryGridOverlay(InventoryGridConfig config, Client client, ItemMana
 		}
 
 		final net.runelite.api.Point mouse = client.getMouseCanvasPosition();
 		final Point mousePoint = new Point(mouse.getX(), mouse.getY());
 
-		for (int i = 0; i < INVENTORY_SIZE; ++i)
+		if (config.showInventoryGrid() && inventoryWidget != null && if1DraggingWidget == inventoryWidget)
+		{
+			for (int i = 0; i < INVENTORY_SIZE; ++i)
+			{
+				WidgetItem widgetItem = inventoryWidget.getWidgetItem(i);
+
+				final Rectangle bounds = widgetItem.getCanvasBounds();
+				boolean inBounds = bounds.contains(mousePoint);
+
+				if (config.showItem() && inBounds)
+				{
+					final WidgetItem draggedItem = inventoryWidget.getWidgetItem(client.getIf1DraggedItemIndex());
+					final BufferedImage draggedItemImage = itemManager.getImage(draggedItem.getId());
+					final int x = (int) bounds.getX();
+					final int y = (int) bounds.getY();
+
+					graphics.setComposite(AlphaComposite.SrcOver.derive(0.3f));
+					graphics.drawImage(draggedItemImage, x, y, null);
+					graphics.setComposite(AlphaComposite.SrcOver);
+				}
+
+				if (config.showHighlight() && inBounds)
+				{
+					graphics.setColor(HIGHLIGHT);
+					graphics.fill(bounds);
+				}
+				else if (config.showInventoryGrid())
+				{
+					graphics.setColor(GRID);
+					graphics.fill(bounds);
+				}
+			}
+		}
+
+		if (config.showBankGrid() && bankWidget != null && draggingWidget != null && draggingWidget.getParent() == bankWidget)
 		{
-			WidgetItem widgetItem = inventoryWidget.getWidgetItem(i);
 
-			final Rectangle bounds = widgetItem.getCanvasBounds();
-			boolean inBounds = bounds.contains(mousePoint);
+			ItemContainer bankInventory = client.getItemContainer(InventoryID.BANK);
+
+			if (bankInventory == null)
+			{
+				return null;
+			}
+
+			Item[] items = bankInventory.getItems();
+			int numItems = items.length;
+			if (numItems == 0)
+			{
+				return null;
+			}
+
+			int currentTab = client.getVar(Varbits.CURRENT_BANK_TAB);
 
-			if (config.showItem() && inBounds)
+			if (currentTab > 0)
 			{
-				final WidgetItem draggedItem = inventoryWidget.getWidgetItem(client.getIf1DraggedItemIndex());
-				final BufferedImage draggedItemImage = itemManager.getImage(draggedItem.getId());
-				final int x = (int) bounds.getX();
-				final int y = (int) bounds.getY();
-
-				graphics.setComposite(AlphaComposite.SrcOver.derive(0.3f));
-				graphics.drawImage(draggedItemImage, x, y, null);
-				graphics.setComposite(AlphaComposite.SrcOver);
+				int startIndex = 0;
+
+				for (int i = currentTab - 1; i > 0; i--)
+				{
+					startIndex += client.getVar(TAB_VARBITS.get(i - 1));
+				}
+
+				int itemCount = client.getVar(TAB_VARBITS.get(currentTab - 1));
+				items = Arrays.copyOfRange(items, startIndex, startIndex + itemCount);
 			}
 
-			if (config.showHighlight() && inBounds)
+			for (int i = 0; i < numItems; ++i)
 			{
-				graphics.setColor(HIGHLIGHT);
-				graphics.fill(bounds);
+
+				Widget widget = bankWidget.getChild(i);
+
+				if (!currentTabContainsWidget(items, widget))
```

Comment:
Is this superfluous if we avoid iterating over the whole bank when we don't need to?

Reply:
Yep.